components/backup-stream/src/endpoint.rs

```diff
@@ -501,23 +534,62 @@ where
                             "failed to register backup stream task {} to router: ranges not found",
                             task.info.get_name()
                         ));
-                        // TODO build a error handle mechanism #error 5
                     }
                 }
             });
         };
     }
 
-    pub fn on_unregister(&self, task: &str) {
+    fn register_service_safepoint(
+        &self,
+        task_name: String,
+        // hint for make optimized safepoint when we cannot get that from `SubscriptionTracker`
+        start_ts: TimeStamp,
+    ) -> impl Future<Output = Result<()>> + Send + 'static {
+        self.pd_client
+            .update_service_safe_point(
+                format!("{}-pause-guard", task_name),
+                self.subs.safepoint().max(start_ts),
+                ReadableDuration::hours(24).0,
+            )
+            .map(|r| r.map_err(|err| err.into()))
+    }
```

Comment:
i feel that this can be implemented more succinctly as an `async fn` (haven't tried if it compiles)

```rust
async fn register_service_safepoint(&self, task_name: &str, start_ts: TimeStamp) -> Result<()> {
    self.pd_client.update_service_safe_point(
        format!("{}-pause-guard", task_name),
        self.subs.safepoint().max(start_ts),
        ReadableDuration::hours(24).0,
    ).await?;
    Ok(())
}
```

Reply:
When make it an async function, I am afraid the returning future may borrow `self` hence may not live `'static` and cannot be `spawn` to the pool. 🤔